lib/ansible/modules/cloud/podman/podman_container.py

```diff
@@ -0,0 +1,1196 @@
+        # and current container to understand if we need to recreate it
+        return True
+
+    @property
+    def running(self):
+        return self.exists and self.info['State']['Running']
+
+    @property
+    def stopped(self):
+        return self.exists and not self.info['State']['Running']
+
+    def get_info(self):
+        rc, out, err = run_podman_command(
+            module=self.module,
+            args=['container', 'inspect', self.name], ignore_errors=True)
+        return json.loads(out)[0] if rc == 0 else {}
+
+    def _perform_action(self, action):
+        command = construct_command_from_params(action, self.module.params)
+        self.module.log("PODMAN-DEBUG: " + " ".join(command))
+        rc, out, err = run_podman_command(
+            module=self.module,
+            args=['container'] + command,
+            ignore_errors=True)
+        if rc != 0:
+            self.module.fail_json(
+                msg="Can't %s container %s" % (action, self.name),
+                stdout=out, stderr=err)
+
+    def run(self):
+        self._perform_action('run')
+
+    def delete(self):
+        self._perform_action('delete')
+
+    def stop(self):
+        self._perform_action('stop')
+
+    def start(self):
+        self._perform_action('start')
+
+    def create(self):
+        self._perform_action('create')
+
+    def recreate(self):
+        self.delete()
+        self.run()
+
+    def restart(self):
+        self.stop()
+        self.run()
+
+
+class PodmanManager:
+
+    def __init__(self, module, results):
+
+        super(PodmanManager, self).__init__()
+
+        self.module = module
+        self.results = results
+        self.name = self.module.params.get('name')
+        self.executable = self.module.get_bin_path(module.params.get('executable'), required=True)
+        self.image = self.module.params.get('image')
+        if self.image:
+            if ensure_image_exists(self.module, self.image):
+                self.results['actions'].append('pulled image %s' % self.image)
+        self.state = self.module.params.get('state')
+        self.restart = self.module.params.get('force_restart')
+        self.recreate = self.module.params.get('recreate')
+        self.container = PodmanContainer(self.module, self.name, self.module.params)
+
+    def update_container_result(self, changed=True):
+        facts = self.container.get_info()
+        self.results.update({'changed': changed, 'container': facts,
+                             'ansible_facts': {'podman_container': facts}})
+        return self.results
+
+    def execute(self):
+        if self.state in ['present', 'started']:
+            if not self.container.exists and not self.image:
+                self.module.fail_json(msg='Cannot create container when image is not specified!')
+            if self.container.running and (self.container.different or self.recreate):
+                if not self.image:
```

Comment:
This check doesn't make sense. You already do this in `main()`

Reply:
yeah, forgot to remove when added check in main, thanks